### PR TITLE
blog post about 'magick' 1.0 release

### PR DESCRIPTION
**Pull request by jeroen**
A new blog post announcing the 1.0 release of magick 🎩 ✨🐇!

To maximize visibility, if we schedule this for `2017-08-15`, please publish this around `2017-08-15 00:00 GMT` time, i.e. Monday 3PM PST.

### CODE REVIEW

**Comment by jeroen** [comment]
Update: post has been updated from comments by @maelle and approved after inclusion of the appropriate emoji.

File: _posts/2017-08-15-magick-10.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+[![collage](http://livefreeordichotomize.com/images/we-r-ladies.jpeg)](http://livefreeordichotomize.com/2017/07/18/the-making-of-we-r-ladies/)
+
+And [Daniel P. Hadley](https://twitter.com/danielphadley/status/884845188979359744) just likes to make confusing graphics:
+
+[![travolta](http://danielphadley.com/images/Cars_Travolta.gif)](http://danielphadley.com/ggplot-Logo/)
+
+Now, 1 year later, the 1.0 release marks an important milestone: the addition of a new native graphics device (which serves as a hybrid between a magick image object and an R plot) bridges the gap between graphics and image processing in R. 
+
+This blog post explains how the magick device allows you to seamlessly combine graphing with image processing in R. You can either use it to post-process your R graphics, or draw on imported images using the native R plotting machinery. We hope that this unified interface will make it easier to beautiful, reproducible images with R.
```
Comment (sckott) [inline _posts/2017-08-15-magick-10.md]: easier to __make/build__ ? beautiful ...

File: _posts/2017-08-15-magick-10.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+## Native Magick Graphics
+
+The `image_graph()` function opens a new graphics device similar to e.g. png() or x11(). It returns an image object to which the plot(s) will be written. Each page in the plotting device will become a frame (layer) in the image object.
```
Comment (sckott) [inline _posts/2017-08-15-magick-10.md]: maybe `png()` and `x11()` in backticks?

File: _posts/2017-08-15-magick-10.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+# Write to file
+image_write(out, "myplot.png")
+```
+
+![out](/assets/blog-images/2017-08-15-magick-10/out.png)
+
+
+## Drawing Device
+
+The `image_draw()` function opens a graphics device to draw on top of an exiting image using pixel coordinates.
```
Comment (sckott) [inline _posts/2017-08-15-magick-10.md]: existing

File: _posts/2017-08-15-magick-10.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+One feature that is already available is the `image_ocr()` function which extracts text from the image using the rOpenSci [tesseract](https://ropensci.org/blog/blog/2016/11/16/tesseract) package. Another cool example of using image analysis is the [collage](https://github.com/ThinkRstat/collage) package which calculates [color histograms](https://github.com/ThinkRstat/collage#histograms) to select appropriate tile images for creating a collage.
+
+As part of supporting supporting analysis tools we plan to extract the bitmap (raster) classes into a separate package. This will enable package authors to write R extensions to analyze and manipulate on the raw image data, without necessary depending on magick. Yet the user can always rely on magick as a powerful toolkit to import/export images and graphics into such low level bitmaps.
```
Comment (sckott) [inline _posts/2017-08-15-magick-10.md]: s/necessary/necessarily ?

**Comment by jeroen** [comment]
Thank you @sckott. All fixed now.

File: _posts/2017-08-15-magick-10.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+For example [Bob Rudis](https://twitter.com/hrbrmstr/status/758304420224466944) uses magick to visualize California drought data from the U.S. Drought Monitor (click on the image to go find out more):
+
+[![drought](https://jeroen.github.io/images/drought.gif)](https://rud.is/b/2016/07/27/u-s-drought-animations-with-the-witchs-brew-purrr-broom-magick/)
+
+R-ladies [Lucy D'Agostino McGowan](http://www.lucymcgowan.com/) and [Maëlle Salmon](http://www.masalmon.eu/) demonstrate how to make a beautiful collage:
+
+[![collage](http://livefreeordichotomize.com/images/we-r-ladies.jpeg)](http://livefreeordichotomize.com/2017/07/18/the-making-of-we-r-ladies/)
+
+And [Daniel P. Hadley](https://twitter.com/danielphadley/status/884845188979359744) just likes to make confusing graphics:
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: "confusing" sounds very negative. Say "entertaining" "creative" "funky"...?

File: _posts/2017-08-15-magick-10.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+[![travolta](http://danielphadley.com/images/Cars_Travolta.gif)](http://danielphadley.com/ggplot-Logo/)
+
+Now, 1 year later, the 1.0 release marks an important milestone: the addition of a new native graphics device (which serves as a hybrid between a magick image object and an R plot) bridges the gap between graphics and image processing in R.
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: s/1/one/

(did I say that right?)

File: _posts/2017-08-15-magick-10.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+tags:
+- r
+- packages
+- software
+- images
+- magick
+---
+
+
+Last week version 1.0 of the [magick](https://cran.r-project.org/web/packages/magick/index.html) package has appeared on CRAN: an ambitious effort to modernize and simplify high quality image processing in R. This R package builds upon the [Magick++ STL](https://www.imagemagick.org/Magick++/STL.html) which exposes a powerful C++ API to the famous ImageMagick library.
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: insert comma, delete "has" like this:

"Last week, version 1.0 of the [magick](https://cran.r-project.org/web/packages/magick/index.html) package appeared on ..."

File: _posts/2017-08-15-magick-10.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+![RStudio Screenshot](https://jeroen.github.io/images/magick.png)
+
+The best place to start learning about magick is the [vignette](https://cran.r-project.org/web/packages/magick/vignettes/intro.html) which gives a brief overview of the overwhelming amount of functionality in this package.
+
+
+## Towards Release 1.0
+
+Last year around this time rOpenSci [announced](https://ropensci.org/blog/blog/2016/08/23/z-magick-release) the first release of the `magick` package: a new powerful toolkit for image reading, writing, converting, editing, transformation, annotation, and animation in R. Since the initial release there have been a several updates with additional functionality, and many useRs have started to discover the power of this package to take visualization in R to the next level.
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: in "there have been a several"
delete "a"

Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: backticks in `magick` here "For example Bob Rudis uses magick"?

File: _posts/2017-08-15-magick-10.md (hunk 6 above)
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: the clause in parentheses makes the sentence hard to follow. Edit to keep the info?

File: _posts/2017-08-15-magick-10.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+[![collage](http://livefreeordichotomize.com/images/we-r-ladies.jpeg)](http://livefreeordichotomize.com/2017/07/18/the-making-of-we-r-ladies/)
+
+And [Daniel P. Hadley](https://twitter.com/danielphadley/status/884845188979359744) just likes to make confusing graphics:
+
+[![travolta](http://danielphadley.com/images/Cars_Travolta.gif)](http://danielphadley.com/ggplot-Logo/)
+
+Now, 1 year later, the 1.0 release marks an important milestone: the addition of a new native graphics device (which serves as a hybrid between a magick image object and an R plot) bridges the gap between graphics and image processing in R. 
+
+This blog post explains how the magick device allows you to seamlessly combine graphing with image processing in R. You can either use it to post-process your R graphics, or draw on imported images using the native R plotting machinery. We hope that this unified interface will make it easier to produce beautiful, reproducible images with R.
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: magick in backticks? or is this a different usage?
Please check whole post for this

File: _posts/2017-08-15-magick-10.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+
+## Porting ImageMagick Commands to R
+
+The magick 1.0 release now has the core image processing functionality that you expect from an image processing package. But there is still a lot of room for improvement to make magick *the* image processing package in R.
+
+A lot of R users and packages currently shell out to ImageMagick command line tools for performing image manipulations. The goal is to support all these operations in the magick package, so that the images can be produced (and reproduced!) on any platform without requiring the user to install additional software.
+
+Note that ImageMagick library is over 26 years old and has accumulated an enormous amount of features in those years. Porting all of this to R is quite a bit of work, for which feedback from users is important. If there is an imagemagick operation that you like to do in R but you can't figure out how, please [open an issue](https://github.com/ropensci/magick/issues) on Github. If the functionality is currently not supported yet, we will try to add it to the next version.
```
Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: s/amount/number/ in "...an enormous amount of features..."

Comment (stefaniebutland) [inline _posts/2017-08-15-magick-10.md]: s/Github/GitHub/

File: _posts/2017-08-15-magick-10.md (hunk 9 above)
Comment (jeroen) [inline _posts/2017-08-15-magick-10.md]: I don't like to overuse the backticks because the pink coloring makes it difficult to read. Therefore I use the backticks only for actual R commands, not to name the package.

**Comment by jeroen** [comment]
Thanks @stefaniebutland all fixed now!

**Comment by stefaniebutland** [comment]
@sckott re: Jeroen's comment above about backticks on commands vs pkg names, can you please remind me of our blog "policy" so I don't steer people wrong?

**Comment by sckott** [comment]
@stefaniebutland i don't think we have policy on that

**Comment by karthik** [comment]
This is a fantastic post! No further edits from me.
